dsl/src/main/java/tech/pegasys/peeps/network/Network.java

```diff
@@ -481,6 +482,16 @@ private Map<Web3ProviderType, Genesis> createGenesis(
                   }
                   extraData = new GenesisExtraDataQbft(validators);
                   break;
+                case QBFT_SMART_CONTRACT:
+                  if (e == Web3ProviderType.BESU) {
+                    genesisConfig = new GenesisConfigQbft(chainId, new BftConfig());
+                  } else {
+                    genesisConfig = new GoQuorumQbftConfig(chainId, new BftConfig());
+                  }
+                  genesisConfig.setSmartContractTransition(
+                      20, "0xb9a219631aed55ebc3d998f17c3840b7ec39c0cc");
```

Comment:
Can we pass these in from the test somehow so it doesn't need to be hard-coded in the Network

Reply:
Not without a major refactoring - the code is tightly coupled to have things fixed and unchanging up front using builders. Consequently it's not currently possible to define transitions after the tests have started, it's also not possible to have different genesis files - they're basically fixed. 

I agree it's not great.